### ISSUE TITLE
fix: lite_ref compilation

### ISSUE THREAD

**Issue by btrautmann** [jinyus/dart_beacon#93]
**Description**

After upgrading to version `0.42.0` I am unable to compile/run tests due to the following:

```console
./../.pub-cache/hosted/pub.dev/lite_ref-0.5.2/lib/src/scoped/scope_widget.dart:75:36: Error: Superclass has no method named 'removeDependent'.
    if (refs == null) return super.removeDependent(dependent);
                                   ^^^^^^^^^^^^^^^
../../.pub-cache/hosted/pub.dev/lite_ref-0.5.2/lib/src/scoped/scope_widget.dart:88:11: Error: Superclass has no method named 'removeDependent'.
    super.removeDependent(dependent);
          ^^^^^^^^^^^^^^^
../../.pub-cache/hosted/pub.dev/lite_ref-0.5.2/lib/src/scoped/scope_widget.dart:75:36: Error: Can't return a value from a void function.
    if (refs == null) return super.removeDependent(dependent);
```

**Steps To Reproduce**

1. Depend on version `^0.42.0`
2. Utilize `state_beacon` in your app
3. Run app/run tests

**Comment by btrautmann** [comment]
> Which version of flutter are you on?

```console
Flutter 3.16.9 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 41456452f2 (7 weeks ago) • 2024-01-25 10:06:23 -0800
Engine • revision f40e976bed
Tools • Dart 3.2.6 • DevTools 2.28.5
```

**Comment by jinyus** [comment]
Could you update to the latest stable release? the method exists:

https://github.com/flutter/flutter/blob/34b454f42dd6f8721dfe43fc7de5d215705b5e52/packages/flutter/lib/src/widgets/framework.dart#L6054

**Comment by jinyus** [comment]
The constraint is set correctly so I'm not sure why it allowed the update:

https://github.com/jinyus/dart_beacon/blob/9074ef11caf861544721ca88b1b1c02b9a044c93/packages/state_beacon/pubspec.yaml#L6-L8

**Comment by btrautmann** [comment]
Bumping to latest stable (3.19.3) fixed the issue.

**Comment by jinyus** [comment]
fixed constraints in v0.43.0

Thanks for finding this.
